ui/js/directives/perf/compare.js

```diff
@@ -21,25 +20,33 @@ treeherder.directive(
                     return !matchText || item.toLowerCase().indexOf(matchText.toLowerCase()) > (-1);
                 }
                 scope.filterTest = function(item) {
-                    return filter(item, scope.testFilter);
+                    return _.any(scope.filter.split(' '), function(matchText) {
+                        return filter(item, matchText);
+                    });
                 };
                 scope.filterPlatform = function(result) {
-                    return filter(result.name, scope.platformFilter) &&
-                        (!scope.showOnlyImportant || result.isMeaningful) &&
-                        (!scope.showOnlyConfident || result.isConfident) &&
-                        (scope.showUnreliablePlatforms || !_.contains(
+                    return _.any(scope.filter.split(' '), function(matchText) {
+                        return filter(result.name, matchText);
+                    });
```

Comment:
Note that this means changing the logic slightly: we'll populate a list of test + platform groupings when the filter text changes, then just display that outright (removing the conditional logic to filter by platform in the .html, file)